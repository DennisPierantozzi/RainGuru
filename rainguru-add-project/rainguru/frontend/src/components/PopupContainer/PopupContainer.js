import React, { Component } from "react";
import Popup from "react-popup";

export default class PopupContainer extends Component {
    constructor(props) {
        super(props);
    }

    /**
     * Renders the content of the Popup
     * @returns the rendered content of the Popup div
     */
    render() {
        return (<div data-testid="popupContainer"><Popup /></div>);
    }
}
