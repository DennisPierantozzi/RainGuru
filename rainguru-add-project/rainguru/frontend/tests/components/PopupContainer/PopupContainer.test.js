import React from "react";
import {render, screen} from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import PopupContainer from "../../../src/components/PopupContainer/PopupContainer";


describe("PopupContainer tests", () => {
    it("matches snapshot", () => {
        const state = renderer.create(<PopupContainer/>).toJSON();
        expect(state).toMatchSnapshot();
    });

    it("renders PopupContainer", () => {
        render(<PopupContainer/>);
        const element = screen.getByTestId("popupContainer");
        expect(element).toBeTruthy();
    });
});