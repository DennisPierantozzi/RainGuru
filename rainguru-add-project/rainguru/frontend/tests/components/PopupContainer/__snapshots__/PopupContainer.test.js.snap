// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupContainer tests matches snapshot 1`] = `
<div
  data-testid="popupContainer"
>
  <div
    className="mm-popup"
  >
    <div
      className="mm-popup__overlay"
      onClick={[Function]}
      role="presentation"
      style={Object {}}
    />
  </div>
</div>
`;
